File: jp_backend/Models/Details/ParkAnimalDetails.cs
using jp_backend.Database.Entities;

namespace jp_backend.Models.Details
{
    public class ParkAnimalDetails
    {
        public Guid Id { get; set; }
        public int AnimalNumber { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Gender { get; set; } = string.Empty;
        public DateTime Birthdate { get; set; }

        // Fremdverweise der Klasse Park-Tier
        public DinosaurHabitatDetails? Habitat { get; set; }
        public DinosaurDetails? DinosaurType { get; set; }
    }
}

File: jp_frontend/frontend/src/app/api/models/park-animal-details.ts
/* tslint:disable */
/* eslint-disable */
import { DinosaurDetails } from './dinosaur-details';
import { DinosaurHabitatDetails } from './dinosaur-habitat-details';
export interface ParkAnimalDetails {
  animalNumber?: number;
  birthdate?: string;
  description?: null | string;
  dinosaurType?: DinosaurDetails;
  gender?: null | string;
  habitat?: DinosaurHabitatDetails;
  id?: string;
  name?: null | string;
}
